__tests__/pages/p/[id].test.jsx

```diff
@@ -64,4 +78,28 @@ describe('Page', () => {
       expect(postName).toBeEmptyDOMElement()
       expect(headingTitle).toBeEmptyDOMElement()
   })
+
+  it('renders a 404 page when props is not found', async () => {
+
+    global.prisma.post.findUniqueOrThrow.mockRejectedValue(
```

Comment:
Due to the nature of the `getServerSideProps` function, jest `toThrow` will not work in this scenario. Under the `getServerSideProps` docs this function is used to render the page with return user data. `toThrow` will need the catch block to have `throw` instead of `return` meaning that it will not return props object. I used `toEqual` to return the error object and was able to render the 404 page too. I included the previous code for reference.

```
it('Throws error when props is not found', async () => {
    global.prisma.post.findUniqueOrThrow.mockImplementation(() => {
      throw new Error('NotFoundError: No User found error')
    })
    const result = getServerSideProps(params)
    mockRouter.push('/pages/p/1')
    render(<Page {...result.props} />)
    await expect(result).rejects.toThrow('NotFoundError: No User found error')
  })
```